### ISSUE TITLE
Clearer documentation of rejectUnauthorized

### ISSUE THREAD

**Issue by candlerb** [passbolt/passbolt_cli#10]
Some of the sample config files have `"rejectUnauthorized": true`

At first it sounds like setting this to `true` is making things *more* secure.  Better to reject things that aren't authorized, right?

But the only documentation seems to be in the [README](https://github.com/passbolt/passbolt_cli/blob/master/README.md#how-to-get-started) file, where it says:

> 2. Allow self signed certificates
>
> ```
> "agentOptions": {
>    "rejectUnauthorized": true
>  }
> ```

This sounds like it is making things *less* secure.  Self-signed certificates are bad, right?

The confusion stems from:

* What certificates are is it talking about - the HTTPS certificate returned by the API server, or something else?
* What is the "agent" that "agentOptions" refers to? (gpgAgent perhaps, or HTTP user agent, or something else?)
* What sort of "authorization" is this referring to?

Bottom line is: if my web server has a valid HTTPS certificate (as it does), should I set `rejectUnauthorized` to `true` or `false`?  Or else, under what other circumstances is it right to set this to either `true` or `false`?

**Comment by stripthis** [comment]
Hi @candlerb, yes rejectUnauthorized is not safe it basically ignore issues with the certificate (authority, not matching names, etc). There are other options that allow finer and safer control for self signed certificate, see. https://github.com/request/request#using-optionsagentoptions

I'll update the doc to point this out.

**Comment by candlerb** [comment]
But is the unsafe value for "rejectUnauthorized" `true` or `false`?

Looking at [nodejs documentation](https://nodejs.org/docs/latest/api/tls.html#tls_tls_createserver_options_secureconnectionlistener) it says:

> If not false the server will reject any connection which is not authorized with the list of supplied CAs. This option only has an effect if requestCert is true. Defaults to true.

So it seems that if you want to allow self-signed certs you'd have to put `"rejectUnauthorized": false` (which makes sense, but is opposite of what README says)

**Comment by candlerb** [comment]
I agree that a pointer to https://github.com/request/request#using-optionsagentoptions would be good.  It shows the 'right' way to deal with a server with self-signed cert (add a `"ca": ...` value with the PEM for the self-signed CA root, rather than not validating the cert at all)

**Comment by stripthis** [comment]
Fixed with 0.3.3
